liquidity-amo/contracts/MasterAMO.sol

```diff
@@ -341,6 +321,51 @@ abstract contract MasterAMO is
         return IERC20Upgradeable(token).balanceOf(address(this));
     }
 
+    // Binary Search Square Root
+    function sqrt(uint256 x) internal pure returns (uint256) {
```

Comment:
Suggest to use OZ's implementation instead to avoid potential issues: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/math/Math.sol#L529.